src/backend/distributed/commands/truncate.c

```diff
@@ -267,9 +267,7 @@ ErrorIfUnsupportedTruncateStmt(TruncateStmt *truncateStatement)
 
 		ErrorIfIllegallyChangingKnownShard(relationId);
 
-		char relationKind = get_rel_relkind(relationId);
-		if (IsCitusTable(relationId) &&
-			relationKind == RELKIND_FOREIGN_TABLE)
+		if (IsCitusTable(relationId) && IsForeignTable(relationId))
 		{
 			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
```

Comment:
I think this HINT is very complex and  could cause weird edge cases. 
```SQL
truncate foreign_table;
ERROR:  truncating distributed foreign tables is currently unsupported
HINT:  Use citus_drop_all_shards to remove foreign table's shards.
Time: 0.413 ms
```

One option could be that we guide towards `undistribute_table` via the HINT? The foreign tables cannot have foreign keys, which simplifies the story. For partitioned tables (once we resolve the issue I noted in the PR), we should be fine with `undistribute -> TRUNCATE -> add back to the metadata`?